core/variant.h

```diff
@@ -416,6 +430,27 @@ class Variant {
 	}
 };
 
+template <typename T, typename std::enable_if<!std::is_same<T,Object>::value && !std::is_base_of<Object,T>::value &&
+																							!std::is_same<T,Variant>::value && !std::is_base_of<Variant,T>::value,
+																							T>::type* = nullptr>
+Variant::Variant(const T* p_ptr)
+{
+	type = _POINTER;
+	_data._const_ptr = p_ptr;
+}
+
+template <typename T, typename std::enable_if<!std::is_same<T,Object>::value && !std::is_base_of<Object,T>::value &&
+																							!std::is_same<T,Variant>::value && !std::is_base_of<Variant,T>::value,
+																							T>::type* = nullptr>
+T* Variant::asPointer() const
```

Comment:
`asPointer` isn't a good name given that we use snake_case in general. Maybe `operator void*` or `as_pointer` would be better?

Reply:
Yup that was my oversight as I use camel case in my own projects, will fix